base/markdown/render/plain.jl

```diff
@@ -51,19 +51,19 @@ function plaininline(io::IO, md...)
 end
 
 plaininline(io::IO, md::Vector) = !isempty(md) && plaininline(io, md...)
-
-plaininline(io::IO, md::Image) = print(io, "![$(md.alt)]($(md.url))")
-
-plaininline(io::IO, s::String) = print(io, s)
-
-plaininline(io::IO, md::Bold) = plaininline(io, "**", md.text, "**")
-
-plaininline(io::IO, md::Italic) = plaininline(io, "*", md.text, "*")
-
+plaininline(io::IO, md::Image) = print(io, "![", plaininline(md.alt), "](",
+                                       md.url, ")")
+plaininline(io::IO, md::Link) = print(io, "[", plaininline(md.text), "](",
+                                      md.url, ")")
+plaininline(io::IO, md::Bold) = print(io, "**", plaininline(md.text), "**")
+plaininline(io::IO, md::Italic) = print(io, "*", plaininline(md.text), "*")
 plaininline(io::IO, md::Code) = print(io, "`", md.code, "`")
-
+plaininline(io::IO, s::String) = print(io, s)
 plaininline(io::IO, x) = writemime(io, MIME"text/plain"(), x)
 
+plaininline(s::String) = s
```

Comment:
Strings are stored in generated markdown trees as `Any["foo"]` though, so this method will almost never be called – kind of a premature optimisation. Not a big deal either way though I guess.